### PR TITLE
fix various depwarns on 0.7 (that break 0.6)

### PR DESCRIPTION
**Pull request by Sacha0**
This pull request fixes those depwarns under 0.7 that break 0.6. Best!

### CODE REVIEW

**Comment by Sacha0** [comment]
Updated CI here as well. Best!

**Comment by vchuravy** [comment]
This looks like something that could be added to Compat.jl, so that it won't break v0.6?

**Comment by Sacha0** [comment]
> This looks like something that could be added to Compat.jl, so that it won't break v0.6?

Good call. That addition should be straightforward. Best!

File: test/test_allgather.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ end
 
 comm = MPI.COMM_WORLD
 
-for typ in (isdefined(:UnionAll) ? Base.uniontypes(MPI.MPIDatatype) : MPI.MPIDatatype.types)
```
Comment (Keno) [inline test/test_allgather.jl]: Should just be `Base.uniontypes(MPI.MPIDatatype)`, since we've dropped pre 0.6 support.

**Comment by andreasnoack** [comment]
I don't think this is relevant anymore

**Comment by Sacha0** [comment]
Indeed! Thanks Andreas! :)

**Pull request closed without merging**